api/comments/serializers.py

```diff
@@ -72,10 +60,56 @@ def update(self, comment, validated_data):
         return comment
 
     def get_target_type(self, obj):
-        object_type = obj._name
-        if not object_type or object_type not in ['comment', 'node']:
+        if isinstance(obj, Node):
+            return 'nodes'
+        elif isinstance(obj, Comment):
+            return 'comments'
+        else:
             raise InvalidModelValueError('Invalid comment target.')
-        return object_type
+
+
+class CommentCreateSerializer(CommentSerializer):
+
+    target_type = ser.SerializerMethodField(method_name='check_target_type')
+
+    def check_target_type(self, obj):
```

Comment:
This is a bit  of a misnomer. `check_target_type` implies that a boolean is being returned. How about `get_validated_target_type`?